src/tokenizer.rs

```diff
@@ -43,11 +44,12 @@ lazy_static! {
     static ref PATTERN_CLOSE_PAREN: Regex = Regex::new(r"^(\))").unwrap();
 
     static ref PATTERN_WHITESPACE: Regex = Regex::new(r"^\s+").unwrap();
+    static ref PATTERN_CHARS_AFTER_NEWLINE: Regex = Regex::new(r"\n([^\n]+)$").unwrap();
 }
 
 /// Tries to matches the given pattern against the string slice.
 /// If it does, the 'tokenizer' fn is invokved to turn the result into a token.
-fn try_advance<'a, F>(source: &'a str, pattern: &Regex, tokenizer: F) -> Option<(&'a str, TokenKind<'a>)>
+fn try_advance<'a, F>(source: &'a str, pattern: &Regex, tokenizer: F) -> Option<(&'a str, &'a str, TokenKind<'a>)>
```

Comment:
This is probably at the point where I'd want to extract out a named struct for this function's return value iinstead of a tuple

Reply:
Yeah, that's probably for the best! I'll pull that out.